main.go

```diff
@@ -107,8 +193,10 @@ func Error() error {
 
 // Stdout stream for the command
 func (c *Cmd) Stdout() string {
-	c.validate()
-	return c.stdout
+	c.validateHasStarted()
+	c.stdout.mu.Lock()
+	defer c.stdout.mu.Unlock()
+	return c.stdout.content
```

Comment:
Do you need a lock for read operations? What is the problem you're trying to prevent?

Reply:
Initially I hadn't added the `retryStringTest` funcyet  and needed to prevent reading the output content while this was still being written to prevent race conditions and unnecessary failures. This seemed to work initially but still ocasionally gave failing tests. 

I decided to add a retry mechanism, as my current approach didn't seem fully reliable. In addition the mutex approach would not work with long running commands that might need longer to reply with the desired output. 

I'm not sure if it's worth keeping around with the retry mechanism, at best it prevents an extra retry attempt because it forces the write to finish before executing the test.